### ISSUE TITLE
Data leakage in train/test split - compositions for one load are in training and for another load in test set

### ISSUE THREAD

**Issue by sgbaird** [ziyan1996/VickersHardnessPrediction#6]
Since this doesn't give as great of a view on the extrapolation performance, it would be interesting to see a group cross-validation scheme via e.g. https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GroupKFold.html where all the data for a particular composition is either entirely in the training set or entirely in the test set.

**Comment by sgbaird** [comment]
These were the results from my comparison of CV with group-CV:
```python
# CV-MAE: 1.67359 (HV)
# GCV-MAE: 2.2957275 (HV)
# CV-RMSE: 2.70621 (HV)
# GCV-RMSE: 3.4197476 (HV)
```
See https://github.com/sparks-baird/VickersHardnessPrediction/blob/master/vickers_hardness/hv_cv.py
